src/daemon/metric.c

```diff
@@ -280,18 +279,19 @@ int metric_identity(strbuf_t *buf, metric_t const *m) {
   if ((buf == NULL) || (m == NULL) || (m->family == NULL)) {
     return EINVAL;
   }
+  label_set_t const *resource = &m->family->resource;
 
   int status = strbuf_print(buf, m->family->name);
-  if (m->resource.num == 0 && m->label.num == 0) {
+  if (resource->num == 0 && m->label.num == 0) {
     return status;
   }
 
   status = status || strbuf_print(buf, "{");
 
   bool first_label = true;
-  if (m->resource.num == 0) {
-    status = status || format_label_set(buf, &m->resource,
-                                        RESOURCE_LABEL_PREFIX, first_label);
+  if (resource->num != 0) {
```

Comment:
Sorry about that!

(I think I had ">" there originally, but force-push caused my pending suggestion to disappear and I messed when quickly re-writing it.  Maybe you could consider leaving force pushes until reviews are about finished?)

Reply:
Oh, sorry about that! Yeah, I can rebase and clean up the commits later in the process, or just accept that my stupid mistakes will be forever enshrined in the commit history ;)